src/openvic-simulation/types/IdentifierRegistry.cpp

```diff
@@ -17,11 +17,11 @@ std::ostream& OpenVic::operator<<(std::ostream& stream, HasIdentifier const* obj
 }
 
 HasColour::HasColour(colour_t new_colour, bool cannot_be_null, bool can_have_alpha) : colour(new_colour) {
-	assert((!cannot_be_null || colour != NULL_COLOUR) && colour <= (!can_have_alpha ? MAX_COLOUR_RGB : MAX_COLOUR_ARGB));
+	assert((!cannot_be_null || colour != colour_t::null) && colour <= (!can_have_alpha ? colour_t::max_rgb : colour_t::max_argb));
```

Comment:
```suggestion
	assert((!cannot_be_null || !colour.is_null()) && (can_have_alpha || colour <= colour_t::max_rgb));
```

The `max_argb` check isn't necessary since `colour_t` can't take greater values. Also a `has_alpha` function for `colour_t` would be useful, it could replace all the existing checks against `colour_t::max_rgb` and avoids potential confusion around which comparison operator to use.